guides/app-guides/agEUR/borrow.md

```diff
@@ -2,58 +2,95 @@
+
+In the case of yield-bearing asset, the tokens sent to the protocol are swapped to the underlyings of the yield-bearing asset, and deposited in the platform they earn yield from. For example, in the case of Curve LP tokens, the underlyings are deposited on Curve, the LP tokens obtained staked on Convex, and used as collateral for a vault.
+
+All external rewards accumulated by the vault's staked collateral can be [claimed](#claiming-your-rewards) directly from the Angle app in one transaction, such that there is **no opportunity cost to use Angle** to stake tokens.
+
+{% hint style="info" %}
+To check out all these steps, click on the `Transaction Details` on the right side of the screen.
+{% endhint %}
 
-When repaying your debt, you have two options: use your vault collateral, or use your debt token. If you choose to repay your debt with your vault collateral, the tokens will be swapped to debt tokens on your behalf and used to repay your debt.
+In the following example, 20 FRAX are deposited in the Curve FRAX/USDC pool, the LP tokens are staked on Stake DAO, and deposited as collateral in the vault. 10 agEUR are borrowed from the vault
 
-Here for example, wETH will be swapped into 1 agEUR to repay part of the vault's debt. 0.01 wETH of the vault's collateral will also be withdrawn.
+![Add/Borrow agEUR](/.gitbook/assets/borrow-lp.png)
 
-![Repay and withdraw](/.gitbook/assets/repay-with-collat.png)
+## Repay your debt or close your vault
+
+Once a vault is created, you can repay part of your debt or close it completely. You can also withdraw some collateral up to the minimum LTV.
+
+When repaying your debt, you can either use your vault collateral balance, your wallet balance, or a mix of both. If you choose to repay your debt with your vault collateral, the tokens are swapped to debt tokens on your behalf and used to repay your debt.
 
 Here are the steps to Repay debt, withdraw collateral, or close your vault:
 
 1. Go to the `Borrow` section of the [app](https://app.angle.money/#/borrow)
 2. Click on the `Repay` button on your vault.
-3. Tick the box above the inputs to repay all your debt and close your vault. You will get all the remaining collateral.
-4. Enter the amount of collateral you want to remove
-5. Enter the amount of stablecoins debt you want to repay in the second input.
-6. Choose whether to use your collateral or your agEUR wallet balance to repay your debt.
+3. Tick the box above the inputs if you want to repay all your debt and close your vault. You will get all the remaining collateral.
+4. Enter the amount of collateral you want to remove and the token you want to receive in your wallet.
+5. Enter the amount collateral you want to use to repay your debt in the bottom left input.
+6. Enter the amount of debt token from your wallet balance you want to use to repay your debt.
 7. Click on `Send` to confirm the transaction.
```

Comment:
We no longer have the `Send` button: update this?

Let's insist on the transaction simulation as well, we need to mention it everywhere possible